### ISSUE TITLE
Bugs with forms published online using Form Builder

### ISSUE THREAD

**Issue by Ne0-d3gen** [formtools/module-submission_history#5]
Hey @benkeen, just noticed you had closed my previous issue so reporting as new issue; 

Errors in some of my submission history tables - **this seems to occur only if a submission is created using the online Form Builder version of the form** - getting the following error messages in the changed by and changed fields columns:

> Notice: Undefined offset: 0 in /home/acutemed/warwickacutemedicine.co.uk/acportfolio/themes/default/cache/8cd9afd56e206bc7f9f33c4146217d879ef53afc_0.file.list_history.tpl.php on line 92
> 
> Notice: Undefined offset: 0 in /home/acutemed/warwickacutemedicine.co.uk/acportfolio/themes/default/cache/8cd9afd56e206bc7f9f33c4146217d879ef53afc_0.file.list_history.tpl.php on line 93

Please click on image below for screenshot:
  
![image](https://user-images.githubusercontent.com/13905479/34471216-5f6ce198-ef3b-11e7-992d-f36fe4f723bf.png)

**Comment by benkeen** [comment]
Thanks @tomheaps!  Don't worry, I hadn't missed your previous post - it's just been a busy couple of days. But it's always most convenient to post them separately.

I'll hopefully take a look tonight, if not tomorrow.

**Comment by benkeen** [comment]
Hi @tomheaps, I was just tinkering around with this. Was the user that made changes to that submission deleted, per chance? I'm able to reproduce that problem when that occurs. The module *should* handle it properly - I'll take a look to see why not.

**Comment by Ne0-d3gen** [comment]
@benkeen, the error is present the first time the form is created / submitted using the online Form Builder version of the form. It happens if you complete a form online with nobody logged in AND if any active user (not deleted) is logged in. The form itself is set to hidden within Form tools for the online view I am using for form builder but there are also admin views and a private view assigned to specific users for viewing the results of submitted forms.

**Comment by benkeen** [comment]
Hey @tomheaps, sorry for the wait on this one - it's been a busy few days. I put in a rather generic patch that *should* fix this and any other scenarios. Mind trying out 2.0.3 and seeing if the problem persists?

**Comment by Ne0-d3gen** [comment]
@benkeen no need to apologize. Have upgraded to 2.03 and can confirm that all pre-existing errors in the submission history table have vanished. I have tested further by creating a submission online both when logged in as a user and when not logged in - everything working perfectly! Thanks once again for your patience.

**Comment by benkeen** [comment]
Thanks @tomheaps!
